src/xhr.js

```diff
@@ -80,7 +84,7 @@ const xhrFactory = function() {
     const xhrMethod = videojs.Vhs.xhr.original === true ? videojsXHR : videojs.Vhs.xhr;
 
     const request = xhrMethod(options, function(error, response) {
-      return callbackWrapper(request, error, response, callback);
+      return callbackWrapper(request, error, response, callback, beforeResponse);
```

Comment:
I believe the `callbackWrapper` gets invoked directly in other places in VHS. That means we likely need a different solution because the user-provided response callback needs to be called on all requests.